httpserver/idp_handlers.go

```diff
@@ -0,0 +1,490 @@
+</body>
+</html>`
+
+	w.Header().Set("Content-Type", "text/html")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write([]byte(html))
+}
+
+// handleIDPLoginSubmit handles login form submission
+func (s *Server) handleIDPLoginSubmit(w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+		s.writeError(w, http.StatusBadRequest, "Failed to parse form")
+		return
+	}
+
+	username := r.FormValue("username")
+	password := r.FormValue("password")
+	redirectURI := r.FormValue("redirect_uri")
+
+	if username == "" || password == "" {
+		s.writeError(w, http.StatusBadRequest, "Username and password required")
+		return
+	}
+
+	// Authenticate user
+	ctx := r.Context()
+	if err := s.idpProvider.storage.AuthenticateUser(ctx, username, password); err != nil {
+		s.logger.Warn(logging.DestinationHTTP, "IDP authentication failed", "username", username)
+		s.writeError(w, http.StatusUnauthorized, "Invalid credentials")
+		return
+	}
+
+	s.logger.Info(logging.DestinationHTTP, "IDP user authenticated", "username", username)
+
+	// Create session for authenticated user
+	// Store username in cookie or session store (using simple cookie for now)
+	// TODO: Set Secure flag to true in production with HTTPS
+	http.SetCookie(w, &http.Cookie{
+		Name:     "idp_session",
+		Value:    username,
+		Path:     "/",
+		HttpOnly: true,
+		Secure:   false, // Set to true when using HTTPS in production
+		SameSite: http.SameSiteLaxMode,
+		MaxAge:   3600, // 1 hour
+	})
+
+	// Redirect back to authorization endpoint or provided redirect_uri
+	if redirectURI != "" {
+		http.Redirect(w, r, redirectURI, http.StatusFound)
+		return
+	}
+
+	// Default redirect to success page
+	w.Header().Set("Content-Type", "text/html")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write([]byte(`<!DOCTYPE html>
+<html>
+<head><title>Login Successful</title></head>
+<body>
+    <h2>Login Successful</h2>
+    <p>You are now logged in as ` + username + `</p>
+</body>
+</html>`))
+}
+
+// handleIDPAuthorize handles OAuth2 authorization requests for IDP
+func (s *Server) handleIDPAuthorize(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	// Check if user is authenticated via session cookie
+	cookie, err := r.Cookie("idp_session")
+	if err != nil || cookie.Value == "" {
+		// User not authenticated, redirect to login with return URL
+		loginURL := "/idp/login?redirect_uri=" + url.QueryEscape(r.URL.String())
+		http.Redirect(w, r, loginURL, http.StatusFound)
+		return
+	}
+
+	username := cookie.Value
+
+	// Create a new authorization request
+	ar, err := s.idpProvider.oauth2.NewAuthorizeRequest(ctx, r)
+	if err != nil {
+		s.logger.Error(logging.DestinationHTTP, "Failed to create authorize request", "error", err)
+		s.idpProvider.oauth2.WriteAuthorizeError(ctx, w, ar, err)
+		return
+	}
+
+	// Create session for the user
+	session := DefaultIDPSession(username)
+
+	// Grant all requested scopes
+	for _, scope := range ar.GetRequestedScopes() {
+		ar.GrantScope(scope)
```

Comment:
If it's not done elsewhere in the PR, limit to the standard OIDC scopes.  Don't blindly approve the client-requested scope.